File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV28.java
package com.hjq.permissions;

import android.app.Activity;
import android.content.Context;

class PermissionDelegateImplV28 extends PermissionDelegateImplV26 {
    PermissionDelegateImplV28() {
    }

    public boolean isGrantedPermission(Context context, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.ACCEPT_HANDOVER)) {
            return PermissionUtils.checkSelfPermission(context, str);
        }
        return super.isGrantedPermission(context, str);
    }

    public boolean isPermissionPermanentDenied(Activity activity, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.ACCEPT_HANDOVER)) {
            return !PermissionUtils.checkSelfPermission(activity, str) && !PermissionUtils.shouldShowRequestPermissionRationale(activity, str);
        }
        return super.isPermissionPermanentDenied(activity, str);
    }
}

File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV28.kt
package com.hjq.permissions

import android.app.Activity
import android.content.Context

internal open class PermissionDelegateImplV28 : PermissionDelegateImplV26() {
    override fun isGrantedPermission(context: Context, permission: String): Boolean {
        if (PermissionUtils.equalsPermission(permission, Permission.ACCEPT_HANDOVER)) {
            return PermissionUtils.checkSelfPermission(context, permission)
        }
        return super.isGrantedPermission(context, permission)
    }

    override fun isPermissionPermanentDenied(activity: Activity, permission: String): Boolean {
        if (PermissionUtils.equalsPermission(permission, Permission.ACCEPT_HANDOVER)) {
            return !PermissionUtils.checkSelfPermission(activity, permission) &&
                !PermissionUtils.shouldShowRequestPermissionRationale(activity, permission)
        }
        return super.isPermissionPermanentDenied(activity, permission)
    }
}
